src/components/Table/nestedCards.tsx

```diff
@@ -28,25 +28,41 @@ type Chrome = () => JSX.Element;
 type ChromeBuffer = Chrome[];
 
 export class NestedCards {
+  private columns: GridColumn<any>[];
   // A stack of the current cards we're showing
   private readonly openCards: Array<string> = [];
   // A buffer of the open/close/spacer rows we need between each content row.
   private readonly chromeBuffer: ChromeBuffer = [];
   private readonly styles: NestedCardsStyle;
 
-  constructor(private columns: GridColumn<any>[], private filteredRows: RowTuple<any>[], private style: GridStyle) {
+  constructor(columns: GridColumn<any>[], style: GridStyle) {
+    this.columns = columns;
     this.styles = style.nestedCards!;
   }
 
-  maybeOpenCard(row: GridDataRow<any>): boolean {
+  /**
+   * Maybe add an opening Chrome row to the given `row` if its a nested card.
+   *
+   * @param row The row which will be opened. The open Chrome row will appear
+   * above this row.
+   * @param buffer The buffer, array of rows, to add the open Chrome row.
+   * Currently there are two buffers, one for header rows (`headerRows`) and a
+   * second for filtered rows (`filteredRows`).
+   */
+  maybeOpenCard(row: GridDataRow<any>, buffer: RowTuple<any>[]): boolean {
     const card = this.styles.kinds[row.kind];
     // If this kind doesn't have a card defined, don't put it on the card stack
     if (card) {
       this.openCards.push(row.kind);
       this.chromeBuffer.push(makeOpenOrCloseCard(this.openCards, this.styles.kinds, "open"));
     }
     // But always close previous cards if needed
-    maybeCreateChromeRow(this.columns, this.filteredRows, this.chromeBuffer);
+    maybeCreateChromeRow({
+      key: `chrome-open-${row.kind}-${row.id}`,
```

Comment:
I was thinking about whether these keys should be called "chrome-open" or "chrome-closed" or what not...because a single chrome row can include both open chrome + spacer chrome + closed chromes, but the key would still be called `chrome-_open_...`...

Kinda thinking, instead of passing `key: ` to `maybeCreateChromeRow` and having that id be based on the row kind/row id we're on, what if the key was just `chrome-${chromeRowIndex++}` where `chromeRowIndex` is just a private variable that starts at 0 and we increment up.

Granted, pretty minor change...

(...I guess `maybeCreateChromeRow` is a top-level function, so we'd have to pass in the `chromeRowIndex`; if it was a method on the `NestedCards` class, it could access the `chromeRowIndex` field internally, which seems nice.)

Reply:
Got it, good thought. Refactoring!